main.go

```diff
@@ -462,3 +487,207 @@ func stopStalePipelines (client *gitlab.Client, vars []*gitlab.PipelineVariable)
 
 	}
 }
+
+// syncIfOSHasEnterpriseRepo detects whether a commit has been merged to
+// the Open Source edition of a repo, and then creates a PR-branch
+// in the Enterprise edition, which is then used in order to open
+// a PR to the Enterprise repo with the OS changes.
+func syncIfOSHasEnterpriseRepo(conf *config, gpr *github.PullRequestEvent) error {
+
+	repo := gpr.GetRepo()
+	if repo == nil {
+		return fmt.Errorf("syncIfOSHasEnterpriseRepo: Failed to get the repository information")
+	}
+
+	// Enterprise repo sentinel
+	switch repo.GetName() {
+	case "deployments":
+	case "useradm":
+	default:
+		log.Debugf("syncIfOSHasEnterpriseRepo: Non-Enterprise repository detected: (%s). Not syncing", repo.GetName())
+		return nil
+	}
+
+	pr := gpr.GetPullRequest()
+	if pr == nil {
+		return fmt.Errorf("syncIfOSHasEnterpriseRepo: Failed to get the pull request")
+	}
+
+	// If the action is "closed" and the "merged" key is "true", the pull request was merged.
+	// While webhooks are also triggered when a pull request is synchronized, Events API timelines
+	// don't include pull request events with the "synchronize" action.
+	if gpr.GetAction() == "closed" && pr.GetMerged() {
+
+		// Only sync on Merges to master or release branches and
+		// verify release branches.
+		branch := pr.GetBase()
+		if branch == nil {
+			return fmt.Errorf("syncIfOSHasEnterpriseRepo: Failed to get the base-branch of the PR: %v", branch)
+		}
+
+		syncBranches := regexp.MustCompile(`(master|[0-9]{1,2}\.[0-9]{1,2}\.x)`)
```

Comment:
I don't really get the regex, now it will not work for Mender release 2.123.x

Yes, that is quite unlikely, but isn't it just simpler to use `[0-9]+`?

Reply:
Sure, `[0-9]+` it is